app/models/geo_local_authority.rb

```diff
@@ -0,0 +1,31 @@
+class GeoLocalAuthority < ApplicationRecord
+  include RGeo::ActiveRecord::GeometryMixin
+
+  SRID = 27_700
+
+  def self.nearest_three_to(string)
+    location = Geocoder.search(string).first
+    by_distance(location.longitude.to_f, location.latitude.to_f).limit(3)
+  end
+
+  def self.by_distance(longitude, latitude)
```

Comment:
How many results does it returns by default? Could it turn out to be expensive as do extensive search?

Reply:
It doesn't really return any results - it returns AR scope. The scope itslef uses some of the postgis magic which is rathere quite performant. I wouldn't expect any delays based on the size of the database in this service